keys.go

```diff
@@ -122,11 +123,39 @@ func (c *Context) makeKeyPair(session *pkcs11Session, privHandle *pkcs11.ObjectH
 		return nil, errNoCkaId
 	}
 
+	var pubHandle *pkcs11.ObjectHandle
+
 	// Find the public half which has a matching CKA_ID
-	pubHandle, err := findKey(session, id, label, uintPtr(pkcs11.CKO_PUBLIC_KEY), &keyType)
+	pubHandle, err = findKey(session, id, label, uintPtr(pkcs11.CKO_PUBLIC_KEY), &keyType)
 	if err != nil {
-		return nil, err
+		p11Err, ok := err.(pkcs11.Error)
+
+		if len(label) == 0 && ok && p11Err == pkcs11.CKR_TEMPLATE_INCONSISTENT {
+			// This probably means we are using a token that doesn't like us passing empty attributes in a template.
```

Comment:
I'm not sure this is a good idea because it means we could return different keys on each invocation of this call because P11 does not guarantee any ordering of results when searching. I think it would be best to instead have an entry in a README that describes edge cases with each of the tokens and encourage users to set labels to non-empty (or default) values and to search either ID or a unique lable.

Reply:
The original code exhibits the same behaviour. To some extent, this is something you just need to know about PKCS#11. Although I concur some README additions wouldn't hurt.